config.go

```diff
@@ -141,6 +147,26 @@ func validateOptions(cfg *config) (err error) {
 		}
 	}
 
+	if cfg.vaultClientCertificate != "" {
+		if exists, _ := fileExists(cfg.vaultClientCertificate); !exists {
+			return fmt.Errorf("the client certificate file: %s does not exist", cfg.vaultClientCertificate)
+		}
+	}
+
+	if cfg.vaultClientPrivateKey != "" {
+		if exists, _ := fileExists(cfg.vaultClientPrivateKey); !exists {
+			return fmt.Errorf("the client private key file: %s does not exist", cfg.vaultClientPrivateKey)
+		}
+	}
+
+	if cfg.vaultClientPrivateKey != "" && cfg.vaultClientCertificate == "" {
+		return fmt.Errorf("you are supplying the client private key, but not the certificate")
+	}
+
+	if cfg.vaultClientPrivateKey == "" && cfg.vaultClientCertificate != "" {
+		return fmt.Errorf("you are supplying the client certificate, but not the private key")
+	}
+
 	if cfg.skipTLSVerify == true && cfg.vaultCaFile != "" {
```

Comment:
Does this: `cfg.skipTLSVerify` not apply to your addition? You might need to check if the skipTLSVerify applies to the feature you're adding.